modin/pandas/series.py

```diff
@@ -357,12 +358,25 @@ def apply(self, func, convert_dtype=True, args=(), **kwds):
             apply_func = "agg"
         else:
             apply_func = "apply"
-            # Add this because it only applies for `apply` specifically.
-            kwds["convert_dtype"] = convert_dtype
-        query_compiler = super(Series, self).apply(func, *args, **kwds)
-        # Sometimes we can return a scalar here
-        if not isinstance(query_compiler, type(self._query_compiler)):
-            return query_compiler
+
+        if isinstance(func, string_types) or is_list_like(func):
+            query_compiler = super(Series, self).apply(func, *args, **kwds)
+            # Sometimes we can return a scalar here
+            if not isinstance(query_compiler, type(self._query_compiler)):
```

Comment:
Should this be moved to after the `else` block if a UDF or lambda returns a scalar?

Reply:
They cannot return scalars. Those functions must be applied element-wise which will always return the `Series` type.